torch/distributed/distributed_c10d.py

```diff
@@ -753,10 +755,35 @@ def recv(tensor,
         return src
 
 
+def async_collective(func):
+    @functools.wraps(func)
+    def wrapped(*args, async_op=False, **kwargs):
+        work = func(*args, **kwargs)
+        if work is None:
+            return
+
+        should_profile = torch.autograd._profiler_enabled()
+        if should_profile:
+            rf = torch.autograd.profiler.record_function(func.__name__)
```

Comment:
Should we somehow distinguish between async and sync calls in the profiling output?